op-batcher/batcher/channel_manager_test.go

```diff
@@ -670,3 +670,39 @@ func TestChannelManager_ChannelOutFactory(t *testing.T) {
 
 	require.IsType(t, &ChannelOutWrapper{}, m.currentChannel.channelBuilder.co)
 }
+
+// TestChannelManager_TxData seeds the channel manager with blocks and triggers the
+// blocks->channels pipeline once without force publish disabled, and once with force publish enabled.
+func TestChannelManager_TxData_ForcePublish(t *testing.T) {
+
+	l := testlog.Logger(t, log.LevelCrit)
+	cfg := newFakeDynamicEthChannelConfig(l, 1000)
+	m := NewChannelManager(l, metrics.NoopMetrics, cfg, defaultTestRollupConfig)
+
+	// Seed channel manager with a block
+	rng := rand.New(rand.NewSource(99))
+	blockA := derivetest.RandomL2BlockWithChainId(rng, 200, defaultTestRollupConfig.L2ChainID)
+	m.blocks = []*types.Block{blockA}
+
+	// Call TxData a first time to trigger blocks->channels pipeline
+	txData, err := m.TxData(eth.BlockID{}, false, false, false)
+	require.ErrorIs(t, err, io.EOF)
+	require.Zero(t, txData.Len(), 0)
+
+	// The test requires us to have something in the channel queue
+	// at this point, but not yet ready to send and not full
+	require.NotEmpty(t, m.channelQueue)
+	require.False(t, m.channelQueue[0].IsFull())
+
+	// Call TxData with force publish enabled
+	txData, err = m.TxData(eth.BlockID{}, false, false, true)
+
+	// Despite no additional blocks being added, we should have tx data:
+	require.NoError(t, err)
+	require.True(t, txData.Len() > 0)
```

Comment:
```suggestion
	require.NotZero(t, txData.Len(), "txData should not be empty")
```
(or `require.Greater`)

Reply:
[37ebcb1](https://github.com/ethereum-optimism/optimism/pull/16837/commits/37ebcb1af630ab3cb0f00c6d04bad27af7afbaa1)